client/src/api/rooms.js

```
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+const BASE_URL = ' http://localhost:5000';
+
+export async function fetchRooms() {
+  try {
+    const res = await axios.get(`${BASE_URL}/rooms`);
+    return res.data;
+  } catch (err) {
+    console.error('Error fetching rooms:', err);
+  }
+}
+
+export async function createRoom(name) {
+  try {
+    const res = await axios.post(`${BASE_URL}/rooms`, { name });
+    return res.data;
+  } catch (err) {
+    console.error('Error fetching rooms:', err);
+  }
+}
+
+export async function updateRoom(id, name) {
+  try {
+    const res = await axios.patch(`${BASE_URL}/rooms/${id}`, { name });
+    return res.data;
+  } catch (err) {
+    console.error('Error fetching rooms:', err);
+  }
+}
+
+export async function deleteRoom(id) {
+  try {
+    await axios.delete(`${BASE_URL}/rooms/${id}`);
+    return true;
+  } catch (err) {
+    console.error('Error fetching rooms:', err);
+  }
```

Consider improving error propagation: instead of only logging inside catch blocks, consider `throw err` after logging (or return a rejected Promise) so caller components can present user-facing errors and take appropriate action. Make the chosen approach consistent across all exported functions in this module.